src/utils/auth.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""S3 connection utility classes and methods."""
+
+import secrets
+import string
+from dataclasses import dataclass
+
+from constants import (
+    AUTHENTICATION_DATABASE_NAME,
+    AUTHENTICATION_TABLE_NAME,
+    DEFAULT_ADMIN_USERNAME,
+    POSTGRESQL_DEFAULT_DATABASE,
+)
+from database import DatabaseConnectionInfo
+from utils.utils import WithLogging
+
+
+@dataclass
+class Authentication(WithLogging):
+    """Class representing authentication."""
+
+    database: DatabaseConnectionInfo
+
+    def create_authentication_table(self) -> bool:
+        """Create authentication table in the authentication database."""
+        self.logger.info("Creating authentication table...")
+        query = f"""
+            CREATE TABLE {AUTHENTICATION_TABLE_NAME} (
```

Comment:
```suggestion
            CREATE TABLE {self.database.table} (
```

Reply:
I've commented about this [here](https://github.com/canonical/kyuubi-k8s-operator/pull/8#discussion_r1570005304)